syntax = "proto3";

package insider.sdk;

enum ContentResultE {
    R_CODE_OK = 0;
    R_CODE_ERROR = 1;
    R_CODE_USER_EXIST = 2;
    R_CODE_USER_TOKEN_TIMEOUT = 3;
}

enum ResponseResult {
    OK = 0;
    ERROR = 1;
}

// 协议响应结果
message SdkResult {
    ResponseResult status_code = 1;  // 响应结果状态码，发送方并不需要
    string code = 2;
}

// 具体业务响应结果
message ContentResult {
    ContentResultE status_code = 1;
    string code = 2;
}

// 结果
message Result {
    SdkResult sdk_result = 1;
    ContentResult content_result = 2;
}

message Footer {
    int32 res = 1;
    Result result = 2; // 返回结果，发送方不需要
}